src/main/java/mods/railcraft/common/blocks/tracks/TrackSwitch.java

```diff
@@ -44,7 +46,7 @@ public IIcon getIcon() {
     @Override
     public int getBasicRailMetadata(EntityMinecart cart) {
         int meta = tileEntity.getBlockMetadata();
-        if (cart != null && isSwitched()) {
+        if (cart != null && (isSwitched() || springingCarts.contains(cart.getPersistentID()))) {
```

Comment:
Also...you may need more granularity than isSwitched(), since isSwitched() also takes the cosmetic counters into account. You probably want to use the switched variable directly.